packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactFragment.kt

```diff
@@ -0,0 +1,225 @@
+    get() = (getActivity()?.getApplication() as ReactApplication?)?.reactHost
+
+  public override fun onCreateView(
+      inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+  ): View? = reactDelegate?.let { it.loadApp(); it.reactRootView }
+
+  public override fun onResume() {
+    super.onResume()
+    if (!disableHostLifecycleEvents) {
+      reactDelegate?.onHostResume()
+    }
+  }
+
+  public override fun onPause() {
+    super.onPause()
+    if (!disableHostLifecycleEvents) {
+      reactDelegate?.onHostPause()
+    }
+  }
+
+  public override fun onDestroy() {
+    super.onDestroy()
+    if (!disableHostLifecycleEvents) {
+      reactDelegate?.onHostDestroy()
+    } else {
+      reactDelegate?.unloadApp()
+    }
+  }
+
+  @Deprecated("Deprecated in Java")
+  public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    @Suppress("DEPRECATION")
+    super.onActivityResult(requestCode, resultCode, data)
+    reactDelegate?.onActivityResult(requestCode, resultCode, data, false)
+  }
+
+  /**
+   * Helper to forward hardware back presses to our React Native Host.
+   *
+   * This must be called via a forward from your host Activity.
+   */
+  public fun onBackPressed(): Boolean = reactDelegate?.onBackPressed() ?: false
+
+  /**
+   * Helper to forward onKeyUp commands from our host Activity. This allows [ReactFragment] to handle
+   * double tap reloads and dev menus.
+   *
+   * This must be called via a forward from your host Activity.
+   *
+   * @param keyCode keyCode
+   * @param event event
+   * @return true if we handled onKeyUp
+   */
+  public fun onKeyUp(keyCode: Int, event: KeyEvent): Boolean =
+    reactDelegate?.shouldShowDevMenuOrReload(keyCode, event) ?: false
+
+  @Deprecated("Deprecated in Java")
+  public override fun onRequestPermissionsResult(
+      requestCode: Int, permissions: Array<String>, grantResults: IntArray
+  ) {
+    @Suppress("DEPRECATION")
+    super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+    permissionListener?.let {
+      if (it.onRequestPermissionsResult(requestCode, permissions, grantResults)) {
+        permissionListener = null
+      }
+    }
+  }
+
+  override fun checkPermission(permission: String, pid: Int, uid: Int): Int =
+    getActivity()?.checkPermission(permission, pid, uid) ?: 0
+
+  override fun checkSelfPermission(permission: String): Int =
+    getActivity()?.checkSelfPermission(permission) ?: 0
+
+  @Suppress("DEPRECATION")
+  override fun requestPermissions(
+      permissions: Array<String>, requestCode: Int, listener: PermissionListener?
+  ): Unit {
+    permissionListener = listener
+    requestPermissions(permissions, requestCode)
+  }
+
+  /** Builder class to help instantiate a ReactFragment. */
+  public class Builder {
+    public var mComponentName: String? = null
+    public var mLaunchOptions: Bundle? = null
+    public var mFabricEnabled: Boolean = false
```

Comment:
can you rename them to don't use the `m` prefix?

Reply:
Done! we might need to mark this change as breaking as the builder is public and also these vars